dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/MeController.java

```diff
@@ -148,61 +155,69 @@
-
-        if ( fields.isEmpty() )
+        if ( fieldsContains( "access", fields ) )
         {
-            fields.addAll( Preset.ALL.getFields() );
+            Access access = aclService.getAccess( user, user );
+            user.setAccess( access );
         }
 
-        CollectionNode collectionNode = fieldFilterService.toCollectionNode( User.class,
-            new FieldFilterParams( Collections.singletonList( user ), fields ) );
-
-        response.setContentType( APPLICATION_JSON_VALUE );
-        setNoStore( response );
-
-        RootNode rootNode = NodeUtils.createRootNode( collectionNode.getChildren().get( 0 ) );
-
-        if ( fieldsContains( "settings", fields ) )
-        {
-            rootNode.addChild( new ComplexNode( "settings" ) ).addChildren(
-                NodeUtils.createSimples(
-                    userSettingService.getUserSettingsWithFallbackByUserAsMap( user, USER_SETTING_KEYS, true ) ) );
-        }
-
-        if ( fieldsContains( "authorities", fields ) )
-        {
-            rootNode.addChild( new CollectionNode( "authorities" ) ).addChildren(
-                NodeUtils.createSimples( user.getUserCredentials().getAllAuthorities() ) );
-        }
-
-        if ( fieldsContains( "programs", fields ) )
-        {
-            rootNode.addChild( new CollectionNode( "programs" ) ).addChildren(
-                NodeUtils.createSimples( programService.getUserPrograms().stream()
-                    .map( BaseIdentifiableObject::getUid )
-                    .collect( Collectors.toList() ) ) );
-        }
-
-        if ( fieldsContains( "dataSets", fields ) )
-        {
-            rootNode.addChild( new CollectionNode( "dataSets" ) ).addChildren(
-                NodeUtils.createSimples( dataSetService.getUserDataRead( user ).stream()
-                    .map( BaseIdentifiableObject::getUid )
-                    .collect( Collectors.toList() ) ) );
-        }
-
-        nodeService.serialize( rootNode, APPLICATION_JSON_VALUE, response.getOutputStream() );
+        MeDto meDto = new MeDto(
+            user.getUid(),
+            user.getUsername(),
+            user.getSurname(),
+            user.getFirstName(),
+            user.getEmployer(),
+            user.getLanguages(),
+            user.getGender(),
+            user.getJobTitle(),
+            user.getCreated().toString(),
+            user.getLastUpdated().toString(),
+            user.getDataViewOrganisationUnits(),
+            user.getFavorites(),
+            user.getSharing(),
+            user.getUserGroupAccesses(),
+            user.getUserAccesses(),
+            user.getGroups(),
+            user.getTranslations(),
+            user.getTeiSearchOrganisationUnits(),
+            user.getOrganisationUnits(),
+            user.getExternalAccess(),
+            user.getDisplayName(),
+            user.getAccess(),
+            user.getName(),
+            user.getEmail(),
+            user.getUserAuthorityGroups(),
```

Comment:
Better to make the `User` a constructor argument and then copy the fields in the constrctor.